waterbox/gpgx/core/cd_hw/cd_cart.h

```diff
@@ -40,7 +40,7 @@
  /* CD compatible ROM/RAM cartridge */
 typedef struct
 {
-  uint8 *area;    // always 16K in bizhawk
+  uint8 area[0x830000]; /* cartridge ROM/RAM area (max. 8MB ROM + Pro Action Replay 128KB ROM / 64KB RAM) */   // always 16K in bizhawk
```

Comment:
Why are we static allocating an extra 8MB of junk at all times now?  Dirty states reduces the perf impact of this, but there's still some perf impact.

Reply:
You should ask ekeeke.  I just tried to get the changes as close as possible to his commit that fixed the issue.  If you think it would be best to change it back, then by all means go ahead.  I want that Demons of Asteborg game to work without introducing breaking changes.

Reply:
ekeeke's answer won't be relevant here, as they had no requirements to adhere to the Waterbox memory model and they were changing code that was already diverged between the two branches; in the upstream commit it's changed from a static 0x840000 to 0x830000, but here it was dynamically allocated.

There's clearly some other mismatch going on between the code, as the original comment mentions that this is holding both cartridge ROM and some other things like Action replay, where as we had an "always 16k" comment.

This PR looks quite sloppy and poorly-researched with that juxtaposition:

```c
uint8 area[0x830000]; // always 16K in bizhawk
```

If I recall correctly, a lot of this was work I did to make things more amenable to the waterbox memory model; the old code punned a single ROM area for a lot of things, but for waterbox it made sense to have the ROM be actually ROM (in the sealed area) to keep state size down.

With lazystates, this may not be as relevant anymore, but if that's our path forward, we'll want to completely and holistically change the memory mapping back to match upstream, not half-change things like here.

Reply:
SRAM change should be fine.  Judging from the number of comments related to sanic and kunkle, someone should test that game on the new code as well.

Reply:
Thank you.  I will remove those changes to cd_cart.h and cd_cart.c and rebuild gpgx.  Test those tases again as well as S&K.

Reply:
Made those changes as above and committed. Should not show the cd_cart.h and cdcart.c, and the gpgx.wbx.gz file changing in size slightly.

Tested those tases as done previously and played some of s&k, sonic 3 w/ knuckles, sonic2 w/ knuckles, sonic 3, sonic 2.